src/tools/generate_workspace/src/main/java/com/google/devtools/build/workspace/GenerateWorkspace.java

```diff
@@ -59,7 +60,13 @@ public static void main(String[] args) {
       return;
     }
 
-    GenerateWorkspace workspaceFileGenerator = new GenerateWorkspace(options.outputDir);
+    if ((options.buildOutputFile.isEmpty() && options.workspaceOutputFile.isEmpty())
+        != options.outputDir.isEmpty()) {
```

Comment:
The possibilities are:

1. `workspace_file` and `build_file` (new behavior, requested in #1520)
2. `output_dir` (current behavior)